src/mssqlProtocolHandler.ts

```diff
@@ -0,0 +1,267 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import * as Utils from "./models/utils";
+import { IConnectionInfo } from "vscode-mssql";
+
+enum Command {
+    connect = "/connect",
+    openConnectionDialog = "/openConnectionDialog",
+}
+
+/**
+ * Handles MSSQL protocol URIs.
+ */
+export class MssqlProtocolHandler {
+    constructor() {}
+
+    /**
+     * Handles the given URI and returns connection information if applicable. Examples of URIs handled:
+     * - vscode://ms-mssql.mssql/connect?server=myServer&database=dbName&user=sa&authenticationType=SqlLogin
+     * - vscode://ms-mssql.mssql/connect?connectionString=Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=myPassword;
+     *
+     * @param uri - The URI to handle.
+     * @returns The connection information or undefined if not applicable.
+     */
+    public handleUri(uri: vscode.Uri): IConnectionInfo | undefined {
+        Utils.logDebug(
+            `[MssqlProtocolHandler][handleUri] URI: ${uri.toString()}`,
+        );
+
+        switch (uri.path) {
+            case Command.connect:
+                Utils.logDebug(
+                    `[MssqlProtocolHandler][handleUri] connect: ${uri.path}`,
+                );
+
+                return this.connect(uri);
+
+            case Command.openConnectionDialog:
+                return undefined;
+
+            default:
+                Utils.logDebug(
+                    `[MssqlProtocolHandler][handleUri] Unknown URI path, defaulting to connect: ${uri.path}`,
+                );
+
+                return this.connect(uri);
+        }
+    }
+
+    /**
+     * Connects using the given URI.
+     *
+     * @param uri - The URI containing connection information.
+     * @returns The connection information or undefined if not applicable.
+     */
+    private connect(uri: vscode.Uri): IConnectionInfo | undefined {
+        return this.readProfileFromArgs(uri.query);
+    }
+
+    /**
+     * Reads the profile information from the query string and returns an IConnectionInfo object.
+     *
+     * @param query - The query string containing connection information.
+     * @returns The connection information object or undefined if the query is empty.
+     */
+    private readProfileFromArgs(query: string): IConnectionInfo | undefined {
+        if (!query) {
+            return undefined;
+        }
+
+        const args = new URLSearchParams(query);
+
+        const connectionString = args.get("connectionString") ?? undefined;
```

Comment:
Is there a reason we can't/shouldn't just map directly to the properties in the IConnectionInfo interface? That way, we have one centralized spot for validating which properties are allowed with which others instead of duplicating logic that reasons about, say, authentication type.

I think the right move here is to just loop over `args.keys()` and if present, set the associated properties in `IConnectionInfo`.

----

If we can't do that for whatever reason and need to grab properties, let's discuss.  We should at least try to employ the connection capabilities STS API: https://github.com/microsoft/vscode-mssql/blob/main/src/connectionconfig/connectionDialogWebviewController.ts#L529

Reply:
If we keep this logic here in any capacity, we definitely need to add tests.  But at first glance, I think we should simplify this away.